Rubberduck.VBEEditor/SafeComWrappers/VBA/VBComponents.cs

```diff
@@ -147,18 +149,25 @@ public void ImportSourceFile(string path)
                 var declarationsStartLine = nonAttributeLines + attributeLines + 1;
                 var correctCodeString = string.Join(Environment.NewLine, codeLines.Skip(declarationsStartLine - 1).ToArray());
 
-                component.CodeModule.Clear();
-                component.CodeModule.AddFromString(correctCodeString);
+                using (var codeModule = component.CodeModule)
+                {
+                    codeModule.Clear();
+                    codeModule.AddFromString(correctCodeString);
+                }
+                component.Dispose();
             }
             else if (ext != ComponentTypeExtensions.FormBinaryExtension)
             {
-                Import(path);
+                using(Import(path)){} //Nothing to do here, except properly disposing the wrapper returned from Import.
```

Comment:
This code smells. If `Import(path)` is itself atomic, then it should clean itself up, not spill over to the caller, no?

Reply:
`Import` returns the component it has imported. It cannot dispose that component as it would potentially make it unusable by the caller. It is just that the calling method here is only interested in the sidebar effects.